web/server/codechecker_server/migrations/config/versions/4db450cf38af_add_extra_product_detail_columns.py

```diff
@@ -0,0 +1,71 @@
+"""Add columns for number of runs and latest run storage
+
+Revision ID: 4db450cf38af
+Revises: 302693c76eb8
+Create Date: 2021-06-28 15:52:57.237509
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = '4db450cf38af'
+down_revision = '302693c76eb8'
+branch_labels = None
+depends_on = None
+
+from alembic import op
+import sqlalchemy as sa
+
+from codechecker_common.logger import get_logger
+from codechecker_server.database import database
+from codechecker_server.database.run_db_model import IDENTIFIER as RUN_META
+from codechecker_web.shared import webserver_context
+
+LOG = get_logger('system')
+
+
+def upgrade():
+    op.add_column(
+        'products',
+        sa.Column(
+            'latest_storage_date',
+            sa.DateTime(),
+            nullable=True))
+
+    op.add_column(
+        'products',
+        sa.Column(
+            'num_of_runs',
+            sa.Integer(),
+            server_default="0",
+            nullable=True))
+
+    try:
+        product_con = op.get_bind()
```

Comment:
Another solution could be if we will set these data after the server starts. That way if we failed to connect to a database during migration or something is not properly set (race condition -> I think it is not possible, because we are changing the number of runs with one query but who knows) we will still be able to set these data to the correct value. What do you think @bruntib?

Reply:
No, I think this is fine during migration, so this data is computed only once. Moreover such errors can happen at server start too.